code/datums/components/surgery_initiator.dm

```diff
@@ -0,0 +1,150 @@
+/**
+  *
+  * Allows parent (obj) to initiate surgeries.
+  *
+  */
+/datum/component/surgery_initiator
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	///allows for post-selection manipulation of parent
+	var/datum/callback/after_select_cb
+
+/datum/component/surgery_initiator/Initialize(_after_select_cb)
+	. = ..()
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+	after_select_cb = _after_select_cb
+
+/datum/component/surgery_initiator/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ITEM_ATTACK, .proc/initiate_surgery_moment)
+
+/datum/component/surgery_initiator/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_ITEM_ATTACK)
+
+
+	/**
```

Comment:
```suggestion
/datum/component/surgery_initiator/UnregisterFromParent()
	UnregisterSignal(parent, COMSIG_ITEM_ATTACK)


/datum/component/surgery_initiator/Destroy()
	if(after_select_cb)
		QDEL_NULL(after_select_cb)
	return ..()


	/**
```
Some callbacks keep references to things in their `object` var and must be cleaned.